### PR TITLE
Update ABI WRT Architecture doc and comments from upstreaming.

### PR DESCRIPTION
**Pull request by jplevyak**
Signed-off-by: John Plevyak <jplevyak@gmail.com>

### CODE REVIEW

File: api/wasm/cpp/proxy_wasm_intrinsics.cc [hunk 1]
```diff
@@ -135,9 +131,12 @@ extern "C" PROXY_WASM_KEEPALIVE void proxy_on_tick(uint32_t root_context_id) {
...
 
-extern "C" PROXY_WASM_KEEPALIVE void proxy_on_create(uint32_t context_id,
-                                                     uint32_t root_context_id) {
-  ensureContext(context_id, root_context_id)->onCreate();
+extern "C" PROXY_WASM_KEEPALIVE void proxy_on_context_create(uint32_t context_id,
+                                                             uint32_t parent_context_id) {
+  if (parent_context_id)
+    ensureContext(context_id, parent_context_id)->onCreate();
+  else
+    ensureRootContext(context_id);
```
Comment (PiotrSikora) [inline api/wasm/cpp/proxy_wasm_intrinsics.cc]: Please add brackets.

Comment (jplevyak) [reply api/wasm/cpp/proxy_wasm_intrinsics.cc]: Done.

File: source/extensions/common/wasm/null/null_plugin.h [hunk 2]
```diff
@@ -47,6 +47,7 @@ namespace Plugin {
 struct NullPluginRegistry {
   uint32_t (*proxy_validate_configuration_)(uint32_t root_context_id,
                                             uint32_t plugin_configuration_size) = nullptr;
+  uint32_t (*proxy_on_context_create_)(uint32_t context_id, uint32_t parent_context_id) = nullptr;
   uint32_t (*proxy_on_start_)(uint32_t root_context_id, uint32_t vm_configuration_size) = nullptr;
```
Comment (PiotrSikora) [inline source/extensions/common/wasm/null/null_plugin.h]: Per the document, this should be renamed to `proxy_on_vm_start()`.

Comment (jplevyak) [reply source/extensions/common/wasm/null/null_plugin.h]: Done.

**Pull request merged**